### PR TITLE
Move database migrations and dev database setup from `poprox-platform`

### PR DESCRIPTION
**Pull request by karlhigley**
This is a full copy of the contents of the `poprox-db` and `dev` directories, with only minor modifications:
* The dev dependencies have been moved from `requirements-dev.txt` into `pyproject.toml`
* The README has been updated to contain instructions for installing the dev dependencies

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,10 +8,16 @@
...
 ## Installation
 
 ```console
 pip install poprox-storage
 ```
+
+### Dev Dependencies
+
+```console
+pip install .[dev]
```
Comment (kluver) [inline README.md:22]: wait. This works? I always do `pip install -r <filename>`

File: dev/dump.sql
Comment (kluver) [inline dev/dump.sql:1]: this is listed as an empty file. I don't think the original version of this was empty. If it was that's a problem on my end.

File: README.md (hunk 1 above)
Comment (kluver) [reply README.md:22]: Ahh -- this works with the changes in pyproject right?

**Review by kluver — APPROVED** [review]
One question which I think is more TIL for kluver than actual uncertainty about correctness.

Another "this file shouldn't be empty" situation which I think is ultimately _fine_ if it is empty -- we can just do a new DB dump if we need to -- I don't think that needs to block this PR.

File: dev/dump.sql
Comment (karlhigley) [reply dev/dump.sql:1]: The version in `poprox-platform` was empty too—I thought that was an intentional placeholder for a dump to be added manually

**Pull request merged**